src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -947,7 +947,11 @@ public void updateAllContentSelectors() {
     @Subscribe
     public void listen(FieldChangedEvent fieldChangedEvent) {
         String newValue = fieldChangedEvent.getNewValue() == null ? "" : fieldChangedEvent.getNewValue();
-        setField(fieldChangedEvent.getFieldName(), newValue);
+        if (SwingUtilities.isEventDispatchThread()) {
```

Comment:
You can, but I thought it made sense to change it directly rather than put it in a queue with possibly strange, despite unlikely, consequences. Edits -> do them (if allowed), gui updates -> put them in the queue.